src/states/elder/mod.rs

```diff
@@ -1558,29 +1579,20 @@ impl Approved for Elder {
         self.send_routing_message(RoutingMessage { src, dst, content })
     }
 
-    fn handle_relocate_event(&mut self, payload: RelocateDetails) -> Result<(), RoutingError> {
+    fn handle_relocate_event(
+        &mut self,
+        payload: RelocateDetails,
+        signature: BlsSignature,
+    ) -> Result<(), RoutingError> {
         info!("{} - handle Relocate: {:?}.", self, payload);
 
         let pub_id = payload.pub_id;
+        let proof = self.chain.prove(&Authority::Section(payload.destination));
```

Comment:
If you want the payload to be checked by both the destination and the relocating node, it might be necessary to create 2 proofs (one for each), or one proof that is the max of both.

SMD will then give you some guarantee that any elder receiving it will be able to validate it, however we might also need to cater for adult eventually and that may require a longer proof for the Adult to verify (don't know how we'll do that but we'll have to track what our adults trust of us).

Reply:
If the proof good for the destination section, doesn't that mean it is also good for the source section, since the source section needs only the latest link in the proof chain anyway?

Reply:
No, there is not such guarantee, (though it is likely that it will be so). 
One source Elder may be lagging behind (and showed up in test), as a result self section message go through the same process of AckMessage with themselves.

For example the destination section may already have sent the AckMessage, but one elder in the source section have yet to reach the new SectionInfo in consensus.